models/classes/TestSessionService.php

```diff
@@ -163,12 +163,11 @@ public function getTestSession(DeliveryExecution $deliveryExecution, $forReading
     {
         $sessionId = $deliveryExecution->getIdentifier();
 
-        //load session if no such session or access mode changed
-        if (
-            !$this->hasTestSession($sessionId) ||
-            (!$forReadingOnly && self::$cache[$sessionId][self::SESSION_PROPERTY_SESSION]->isReadOnly())
-        ) {
+        if (!$this->hasTestSession($sessionId)) {
             $this->loadSession($deliveryExecution, $forReadingOnly);
+        } elseif ($this->accessChangedToReadOnly($forReadingOnly, $sessionId)) {
```

Comment:
Looks like the name of the `accessChangedToReadOnly` is not precise. It should be something like: `accessModeChandedToWrite`

Reply:
Fixed.